debezium-connector-oracle/src/test/java/KafkaDatabaseHistoryTest.java

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import static org.fest.assertions.Assertions.assertThat;
+
+import java.io.File;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.Map;
+
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import io.debezium.config.Configuration;
+import io.debezium.connector.oracle.OracleConnectorConfig;
+import io.debezium.connector.oracle.OracleOffsetContext;
+import io.debezium.connector.oracle.OraclePartition;
+import io.debezium.connector.oracle.Scn;
+import io.debezium.connector.oracle.antlr.OracleDdlParser;
+import io.debezium.connector.oracle.util.TestHelper;
+import io.debezium.doc.FixFor;
+import io.debezium.kafka.KafkaCluster;
+import io.debezium.pipeline.source.snapshot.incremental.SignalBasedIncrementalSnapshotContext;
+import io.debezium.pipeline.spi.Offsets;
+import io.debezium.pipeline.spi.Partition;
+import io.debezium.pipeline.txmetadata.TransactionContext;
+import io.debezium.relational.Column;
+import io.debezium.relational.RelationalDatabaseConnectorConfig;
+import io.debezium.relational.Table;
+import io.debezium.relational.TableId;
+import io.debezium.relational.Tables;
+import io.debezium.relational.ddl.DdlParser;
+import io.debezium.relational.history.DatabaseHistory;
+import io.debezium.relational.history.DatabaseHistoryMetrics;
+import io.debezium.relational.history.HistoryRecord;
+import io.debezium.relational.history.KafkaDatabaseHistory;
+import io.debezium.relational.history.TableChanges;
+import io.debezium.util.Collect;
+import io.debezium.util.Testing;
+
+import oracle.jdbc.OracleTypes;
+
+/**
+ * Unit tests for Oracle's database history.
+ * 
+ * @author Chris Cranford
+ */
+public class KafkaDatabaseHistoryTest {
```

Comment:
Do we need Kafka database history? Why it would not work with MemoryDatabaseHistory? Also the functionality is for the core code so maybe the test should go to the core as I can imagine other connectors like MySQL to support  storing previous table id - see question regarding `RENAME` event.

Reply:
It should work with any history implementation, I just mirrored my work off a similar setup in MySQL.  I'll look at pulling the core part of the test up and leaving the Oracle-bits as a concrete test class as we've done for Incremental Snapshots, etc.